### PR TITLE
issue: 848172 Second Fix - csbuild report error in ring accessing hash_map

### PR DESCRIPTION
**Pull request by DanielLibenson**
(CLANG_WARNING) warning: Assigned value is garbage or undefined

Signed-off-by: Daniel Libenson <daneilli@mellanox.com>

### CODE REVIEW

File: src/vma/dev/ring.h [hunk 1]
```diff
@@ -81,23 +81,23 @@ typedef enum {
 /* udp uc key, only by destination port as we already know the rest */
 typedef struct __attribute__((packed)) flow_spec_udp_uc_key_t {
 	in_port_t 	dst_port;
-	flow_spec_udp_uc_key_t(): dst_port(INPORT_ANY){}//Default constructor
-	flow_spec_udp_uc_key_t(in_port_t d_port): dst_port(d_port){}//Constructor
+	flow_spec_udp_uc_key_t(){ memset(this, 0, sizeof(*this)); }//Default constructor
+	flow_spec_udp_uc_key_t(in_port_t d_port): dst_port(d_port) {}//Constructor
```
Comment (rosenbaumalex) [inline src/vma/dev/ring.h]: @DanielLibenson Don't you need to add the `memset(this, 0, sizeof(*this))` on all CTOR's?

File: src/vma/dev/ring.h [hunk 2]
```diff
@@ -81,23 +81,23 @@ typedef enum {
 /* udp uc key, only by destination port as we already know the rest */
 typedef struct __attribute__((packed)) flow_spec_udp_uc_key_t {
 	in_port_t 	dst_port;
-	flow_spec_udp_uc_key_t(): dst_port(INPORT_ANY){}//Default constructor
-	flow_spec_udp_uc_key_t(in_port_t d_port): dst_port(d_port){}//Constructor
+	flow_spec_udp_uc_key_t(){ memset(this, 0, sizeof(*this)); }//Default constructor
```
Comment (rosenbaumalex) [inline src/vma/dev/ring.h]: Formatting comment: please align spaces for all functions....   `_key_t() { memse...(); }`
some have space and some are missing the extra space.

File: src/vma/dev/ring.h (hunk 1 above)
Comment (DanielLibenson) [inline src/vma/dev/ring.h]: It's not needed, but I will add it for the readability.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/419/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/433/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/438/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/424/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/439/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/425/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/440/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/426/ for details (Mellanox internal link).

File: src/vma/dev/ring.h [hunk 3]
```diff
@@ -80,27 +80,56 @@ typedef enum {
...
-	in_addr_t	dst_ip;
-	in_port_t	dst_port;
-	flow_spec_udp_mc_key_t(): dst_ip(INADDR_ANY), dst_port(INPORT_ANY){}//Default constructor
-	flow_spec_udp_mc_key_t(in_addr_t d_ip, in_addr_t d_port): dst_ip(d_ip), dst_port(d_port){}//Constructor
+  in_addr_t	dst_ip;
+  in_port_t	dst_port;
+
+  /* Just for silencing coverity */
+  flow_spec_udp_mc_key_t () {
+    memset(this, 0, sizeof(*this));
```
Comment (rosenbaumalex) [inline src/vma/dev/ring.h]: will calling `flow_spec_udp_mc_key_t (INADDR_ANY, INPORT_ANY)` work as well?
this can reduce amount of duplicated code.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/442/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/427/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/443/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/428/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/444/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
@DanielLibenson please rebase

**Comment by OphirMunk** [comment]
Merged https://github.com/Mellanox/libvma/pull/191

**Pull request closed without merging**